### ISSUE TITLE
Upon focusing and blurring a required field there is no response

### ISSUE THREAD

**Issue by tomdye** [dojo/widgets#460]
**Bug** 

As per dojo 1.x widgets, I would expect an empty blurred required field to highlight red and show a missing required field message. Currently this does not happen.

![screenshot 2018-02-02 14 59 14](https://user-images.githubusercontent.com/814453/35739483-a75f37d8-0829-11e8-9096-a01cdaf3a709.png)

**Comment by smhigley** [comment]
There are two separate parts to this:

1. **Validating on blur.** The widget itself doesn't control when it validates, and I'm not sure it should. Different designers and developers have different opinions on when validation should happen, and I think it's best left fully controlled. We should probably implement this in examples, though.

2. **Displaying an error message.** This seems related to #22. It's definitely something we should implement, but it might not make it to RC.

**Comment by tomdye** [comment]
I agree that this widget being controlled makes sense, however, I feel that the basic `required` validation could be done within the widget itself in the event that the widget is set as `required`, but does not have an `onValidate` or similar callback passed.

**Comment by smhigley** [comment]
It actually does receive basic `required` validation without `onValidate`, through the magic of native validation :)

The problem is there isn't any default or norm for when a widget should validate. Some people like it on blur, others think the browser default of `onsubmit` is better. I don't think this is somewhere we should be opinionated.

**Comment by tomdye** [comment]
This could be done as part by adding native validation to inputs

**Comment by tomdye** [comment]
closing as outdated
